src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenFunctionPointersTests.cs

```diff
@@ -4271,13 +4575,33 @@ private static void VerifyFunctionPointerSymbol(TypeSymbol type, CallingConventi
             Assert.Equal(expectedConvention, signature.CallingConvention);
 
             Assert.Equal(returnVerifier.RefKind, signature.RefKind);
+            switch (signature.RefKind)
+            {
+                case RefKind.Out:
+                    Assert.True(false, $"Cannot have a return ref kind of {signature.RefKind}");
+                    break;
+
+                case RefKind.RefReadOnly:
+                    Assert.True(CustomModifierUtils.HasInAttributeModifier(signature.RefCustomModifiers));
+                    break;
+            }
             returnVerifier.TypeVerifier(signature.ReturnType);
 
             Assert.Equal(argumentVerifiers.Length, signature.ParameterCount);
             for (int i = 0; i < argumentVerifiers.Length; i++)
             {
                 Assert.Equal(argumentVerifiers[i].RefKind, signature.Parameters[i].RefKind);
                 argumentVerifiers[i].TypeVerifier(signature.Parameters[i].Type);
+                switch (signature.RefKind)
```

Comment:
>switch (signature.RefKind) [](start = 16, length = 26)

What are we testing here? How is it related to parameters?

Reply:
We're ensuring that parameters marked as `in` or `out` have the appropriate metadata.

---
In reply to: [409819988](https://github.com/dotnet/roslyn/pull/43355#discussion_r409819988) [](ancestors = 409819988)

Reply:
> We're ensuring that parameters marked as in or out have the appropriate metadata.

Is this really what is happening? What are signature.RefKind, signature.RefCustomModifiers, signature.RefCustomModifiers? How are they related to parameters?

---
In reply to: [409838517](https://github.com/dotnet/roslyn/pull/43355#discussion_r409838517) [](ancestors = 409838517,409819988)